input4linux-0.8.0-rc2.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+set -euo pipefail
+
+# Configuration
+TEST_MODE="${TEST_MODE:-true}"
+URL="https://github.com/worklouder/input-releases/releases/download/v0.8.0-rc.2/input-Setup-0.8.0-rc.2.exe"
+FILENAME="input-Setup-0.8.0-rc.2.exe"
+DOWNLOAD_DIR="./input_download"
+EXTRACT_DIR="./input_extracted"
+REBUILD_DIR="./input_rebuild"
+APP_64_EXTRACT_DIR="$REBUILD_DIR/app-64"
+FINAL_APP_DIR="./input-app"
+ELECTRON_VERSION="29.2.0"
+
+# Ensure required tools are available
+for cmd in curl 7z asar npm; do
+    if ! command -v "$cmd" &>/dev/null; then
+        echo "Missing required command: $cmd"
+        [[ "$TEST_MODE" == true ]] && echo "Test mode enabled, continuing..." || exit 1
+    fi
+done
+
+# Error handling
+handle_error() {
+    local message="$1"
+    if [[ "$TEST_MODE" == true ]]; then
+        echo "$message (ignored due to TEST_MODE)"
+    else
+        echo "$message"
+        exit 1
+    fi
+}
+
+# Prepare directories
+mkdir -p "$DOWNLOAD_DIR" "$EXTRACT_DIR" "$REBUILD_DIR" "$APP_64_EXTRACT_DIR"
+
+# Download EXE
+echo "Downloading $FILENAME..."
+if ! curl -L "$URL" -o "$DOWNLOAD_DIR/$FILENAME"; then
+    handle_error "Download failed"
+fi
+
+# Extract EXE
```

Comment:
extract only if it has not already been extracted which would prevent redoing slow steps:
```
if [[ ! -d "$APP_64_EXTRACT_DIR" ]]; then
    7z x "$REBUILD_DIR/app-64.7z" -o"$APP_64_EXTRACT_DIR" || handle_error "Extraction failed"
fi
```